src/test/java/com/solvd/LoginTest.java

```diff
@@ -1,29 +1,57 @@
-import com.solvd.pages.common.ProductPageBase;
+import com.solvd.pages.common.ProductListPageBase;
 import org.testng.Assert;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
-public class LoginTest extends AbstractTest{
-    @Test
-    public void testCorrectLogin() {
-        LoginPageBase loginPage = initPage(driver, LoginPageBase.class);
+import java.lang.reflect.Method;
 
-        loginPage.enterName("standard_user");
-        loginPage.enterPassword("secret_sauce");
-        ProductPageBase productPage = loginPage.clickLogin();
-        Assert.assertTrue(productPage.checkIfTitleDisplayed(), "Failed to display product page title");
+public class LoginTest extends BaseTest {
+    //TC001
+    @Test (dataProvider = "data-provider")
+    public void testLoginUser(String name, String password) {
+        LoginPageBase loginPage = initPage(getDriver(), LoginPageBase.class);
+        ProductListPageBase productPage = loginPage.login(name, password);
+        Assert.assertTrue(productPage.isTitlePresent(), "Failed to display product page title");
     }
 
-    @Test
-    public void testInvalidLogin() {
-        LoginPageBase loginPage = initPage(driver, LoginPageBase.class);
-
-        loginPage.enterName("aaa");
-        loginPage.enterPassword("aaa");
-        loginPage.clickLoginExpectingError();
-        Assert.assertTrue(loginPage.checkIfErrorDisplayed(), "Failed to display error");
+    //TC002
+    @Test (dataProvider = "data-provider")
+    public void testInvalidLogin(String name, String password) {
+        LoginPageBase loginPage = initPage(getDriver(), LoginPageBase.class);
+        ProductListPageBase productPage = loginPage.login(name, password);
+        Assert.assertTrue(loginPage.isErrorMessagePresent(), "Failed to display error");
         Assert.assertEquals(loginPage.getErrorText(), "Username and password do not match any user in this service.", "Wrong error message");
     }
+
+    //TC010
+    @Test (dataProvider = "data-provider")
+    public void testLockedOutLogin(String name, String password) {
+        LoginPageBase loginPage = initPage(getDriver(), LoginPageBase.class);
+        ProductListPageBase productPage = loginPage.login(name, password);
+        Assert.assertTrue(loginPage.isErrorMessagePresent(), "Failed to display error");
+        Assert.assertEquals(loginPage.getErrorText(), "Sorry, this user has been locked out.", "Wrong error message");
+    }
+
+    //TC007
+    @Test (dataProvider = "data-provider")
+    public void testLogoutUser(String name, String password) {
+        LoginPageBase loginPage = initPage(getDriver(), LoginPageBase.class);
+        ProductListPageBase productPage = loginPage.login(name, password);
+        loginPage = productPage.logout();
+        Assert.assertTrue(loginPage.isLoginPresent(), "Failed to display login page");
+    }
+
+    @DataProvider (name = "data-provider")
+    public Object[][] dpMethod (Method m){
```

Comment:
You misunderstood how the @DataProvider works. You use a single @DataProvider named "data-provider" across multiple test cases, each expecting different input data. 
It's not scalable you add a case for every test. 

@DataProvider(name = "loginData")
    public Object[][] loginData() {
        return new Object[][]{
                {STANDARD_USER, true},
                {USER_INVALID_CREDENTIALS, false},
        };
    }